src/serve_data.js

```diff
@@ -168,139 +168,178 @@ export const serve_data = {
+
+          if (Number.isInteger(x) && Number.isInteger(y)) {
+            const intX = parseInt(req.params.x, 10);
+            const intY = parseInt(req.params.y, 10);
 
             if (
-              z < item.tileJSON.minzoom ||
-              0 ||
-              x < 0 ||
-              y < 0 ||
-              z > item.tileJSON.maxzoom ||
-              x >= Math.pow(2, z) ||
-              y >= Math.pow(2, z)
+              z < tileJSON.minzoom ||
+              z > tileJSON.maxzoom ||
+              intX < 0 ||
+              intY < 0 ||
+              intX >= Math.pow(2, z) ||
+              intY >= Math.pow(2, z)
             ) {
               return res.status(404).send('Out of bounds');
             }
-
-            xy = [x | 0, y | 0];
-            tileCenter = mercator.bbox(x, y, z);
+            xy = [intX, intY];
+            tileCenter = new SphericalMercator().bbox(intX, intY, z);
           } else {
-            //console.log("float")
-
             if (
-              z < item.tileJSON.minzoom ||
+              z < tileJSON.minzoom ||
+              z > tileJSON.maxzoom ||
               x < -180 ||
               y < -90 ||
-              z > item.tileJSON.maxzoom ||
               x > 180 ||
               y > 90
             ) {
               return res.status(404).send('Out of bounds');
             }
 
-            x = parseFloat(x);
-            y = parseFloat(y);
             tileCenter = [y, x, y + 0.1, x + 0.1];
-            xy = mercator.xyz(tileCenter, z);
-            xy = [xy.minX, xy.minY];
+            const { minX, minY } = new SphericalMercator().xyz(tileCenter, z);
+            xy = [minX, minY];
           }
 
-          //console.log(tileCenter + " ## " + xy);
-          item.source.getTile(z, xy[0], xy[1], async (err, data, headers) => {
-            if (err) {
-              if (/does not exist/.test(err.message)) {
-                return res.status(204).send();
-              } else {
-                return res
-                  .status(500)
-                  .header('Content-Type', 'text/plain')
-                  .send(err.message);
+          let data;
+          if (sourceType === 'pmtiles') {
+            const tileinfo = await getPMtilesTile(source, z, x, y);
+            if (!tileinfo?.data) return res.status(204).send();
+            data = tileinfo.data;
+          } else {
+            data = await new Promise((resolve, reject) => {
+              source.getTile(z, xy[0], xy[1], (err, tileData) => {
+                if (err) {
+                  return /does not exist/.test(err.message)
+                    ? resolve(null)
+                    : reject(err);
+                }
+                resolve(tileData);
+              });
+            });
+          }
+          if (data == null) return res.status(204).send();
+          if (!data) return res.status(404).send('Not found');
+          if (tileJSON.format === 'pbf')
+            return res.status(400).send('Invalid format');
```

Comment:
I know this is my code, but i thought i removed this. There is a check above that check format is either png or webp, so this isn't needed

Reply:
Removed.